### PR TITLE
feat: add tracing support

### PR DESCRIPTION
**Pull request by musale**
Part of https://github.com/microsoft/kiota/issues/1872

### CODE REVIEW

**Comment by baywet** [comment]
@musale Thanks for starting this. Would you mind fixing the formatting (see CI) before we review it?

File: kiota_abstractions/request_information.py [hunk 1]
```diff
@@ -155,18 +159,26 @@ def set_content_from_parsable(
...
+        span = self.start_tracing_span("set_content_from_parsable")
+        try:
+            writer = self._get_serialization_writer(
+                request_adapter, content_type, values, span
+            )
+
+            if isinstance(values, list):
+                writer.write_collection_of_object_values(None, values)
+            else:
+                writer.write_object_value(None, values)
```
Comment (samwelkanda) [inline kiota_abstractions/request_information.py:171]: And here `span.set_attribute(REQUEST_TYPE_KEY, "object")`

File: kiota_abstractions/request_information.py [hunk 2]
```diff
@@ -155,18 +159,26 @@ def set_content_from_parsable(
...
 
-        self._set_content_and_content_type_header(writer, content_type)
+            self._set_content_and_content_type_header(writer, content_type)
+        finally:
```
Comment (samwelkanda) [inline kiota_abstractions/request_information.py:174]: Not needed if we go the context manager way.

File: kiota_abstractions/request_information.py [hunk 3]
```diff
@@ -176,34 +188,46 @@ def set_content_from_scalar(
...
+                request_adapter, content_type, values, span
+            )
+
+            if isinstance(values, list):
+                writer.writer = writer.write_collection_of_primitive_values(
+                    None, values
+                )
             else:
-                raise Exception(f"Encountered an unknown type during serialization {value_type}")
+                value_type = type(values)
```
Comment (samwelkanda) [inline kiota_abstractions/request_information.py:202]: `span.set_attribute(REQUEST_TYPE_KEY, value_type)`

File: kiota_abstractions/request_information.py [hunk 4]
```diff
@@ -3,34 +3,38 @@
...
 from .request_option import RequestOption
 from .serialization import Parsable, SerializationWriter
+from ._version import VERSION
 
 if TYPE_CHECKING:
     from .request_adapter import RequestAdapter
 
 Url = str
 T = TypeVar("T", bound=Parsable)
+OBSERVABILITY_TRACER_NAME = "com.microsoft.kiota:microsoft-python-kiota-abstractions"
```
[the remark this answers is not in the document]
Comment (musale) [reply kiota_abstractions/request_information.py:21]: Without the `com.microsoft.kiota` part?

File: kiota_abstractions/request_information.py [hunk 5]
```diff
@@ -155,18 +159,26 @@ def set_content_from_parsable(
             content_type (Optional[str]): the content type.
             values (Union[T, List[T]]): the models.
         """
-        writer = self._get_serialization_writer(request_adapter, content_type, values)
-
-        if isinstance(values, list):
-            writer.write_collection_of_object_values(None, values)
-        else:
-            writer.write_object_value(None, values)
+        span = self.start_tracing_span("set_content_from_parsable")
```
[the remark this answers is not in the document]
Comment (musale) [reply kiota_abstractions/request_information.py:162]: Initially I used this approach. I can't remember why I changed from it. However, I don't see any issues with using the ctx manager so reverting.

File: kiota_abstractions/request_information.py (hunk 1 above)
Comment (musale) [reply kiota_abstractions/request_information.py:171]: What's the actual string for REQUEST_TYPE_KEY?

Comment (samwelkanda) [reply kiota_abstractions/request_information.py:170]: "com.microsoft.kiota.request.type" seems to be the most common value across the sdks

**Review by samwelkanda — COMMENTED** [review]
Looks good! 
Please add a changelog entry and bump the package version.

**Review by samwelkanda — APPROVED** [review]

**Pull request merged**